index.js

```diff
@@ -57,28 +57,42 @@ AlmondPlatform.prototype.addAccessory = function(device) {
         services.push(Service.Switch);
     }
 
+    if (device.props.SwitchBinary1 !== undefined) {
+        services.push(Service.Switch);
+    }
+
     if (services.length === 0) {
         this.log("Not supported: %s [%s]", device.name, device.type);
         return;
     }
 
-    this.log("Found: %s [%s]", device.name, device.type);
+    for(var id in device._deviceValues){
+        if (device._deviceValues.hasOwnProperty(id)) {
+            if(id != device.props.SwitchBinary && id != device.props.SwitchBinary1 && id != device.props.SwitchBinary2) {
+                return; //This particular device value in the device is not a switch but some other unsupported type
```

Comment:
@pablopoo Hmm, but at this point the only services that are being registered are switches. All this new line is doing is dropping the values that isn't a switch. So if a device has a dimmer and a switch, for example, it'll drop the dimmer, since this code doesn't yet have support for that service, and retain the switch. Or am I missing something? I don't have a device that has more than just a switch other than one with a power meter (which works with this PR right now), so I can't really test those right now.